core/src/main/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassMonitor.scala

```diff
@@ -0,0 +1,101 @@
+    openTelemetry.incrementCounter("fastpass-monitor-receive-msg", tags = openTelemetryTags).unsafeToFuture()
+    deleteExpiredGrants()
+  }
+
+  private def deleteExpiredGrants(): Future[ReadWriteAction[Unit]] =
+    dataSource.inTransaction { dataAccess =>
+      dataAccess.fastPassGrantQuery
+        .findExpiredFastPassGrants()
+        .flatMap { expiredGrants =>
+          logger.info(s"Found ${expiredGrants.size} total expired grants")
+          // Convert to stream and group by workspaceId
+          DBIO.sequence(expiredGrants.to(LazyList).groupBy(_.workspaceId).map { case (workspaceId, workspaceGrants) =>
+            logger.info(s"Found ${workspaceGrants.size} expired grants for workspace $workspaceId")
+            // Query workspace info then group by accountEmail
+            dataAccess.workspaceQuery.findByIdOrFail(workspaceId).flatMap { workspace =>
+              DBIO.sequence(workspaceGrants.groupBy(_.accountEmail).map { case (_, accountEmailGrants) =>
+                // Remove grants for a given workspace and accountEmail
+                removeGrantsForAccountEmailInWorkspace(dataAccess, workspace, accountEmailGrants)
+              })
+            }
+          })
+        }
+        .map(_ => DBIO.successful(()))
+    }
+
+  private def removeGrantsForAccountEmailInWorkspace(dataAccess: DataAccess,
+                                                     workspace: Workspace,
+                                                     grants: Iterable[FastPassGrant]
+  ): ReadWriteAction[Unit] = {
+    // Projects and buckets need different api calls, so group by resourceType
+    grants.groupBy(_.resourceType).map { case (resourceType, resourceTypeGrants) =>
+      val organizationRoles = resourceTypeGrants.map(_.organizationRole).toSet
+      // The grouped resourceTypeGrants are the same except for roles, so we can just take the first one
+      val resourceTypeGrant = resourceTypeGrants.head
+      // For a given resource type, remove the roles for the accountEmail
+      resourceType match {
+        case IamResourceTypes.Project =>
+          googleIamDao.removeRoles(GoogleProject(workspace.googleProjectId.value),
+                                   WorkbenchEmail(resourceTypeGrant.accountEmail.value),
+                                   resourceTypeGrant.accountType,
+                                   organizationRoles
+          )
+        case IamResourceTypes.Bucket =>
+          googleStorageDao.removeIamRoles(
+            GcsBucketName(workspace.bucketName),
+            WorkbenchEmail(resourceTypeGrant.accountEmail.value),
+            resourceTypeGrant.accountType,
+            organizationRoles,
+            userProject = Some(GoogleProject(workspace.googleProjectId.value))
+          )
+        case _ => throw new RuntimeException(s"Unsupported resource type ${resourceTypeGrant.resourceType}")
+      }
+      openTelemetry.incrementCounter("fastpass-monitor-google-call", tags = openTelemetryTags).unsafeToFuture()
```

Comment:
We may want to have separate counters for Project IAM and Bucket IAM.  I am thinking about quotas and trying to track down if one or the other of these starts causing us grief.